### PR TITLE
fix: 修复当依赖包 package.json 内有 exports 字段，并且里面不包含 ./package.json 时 build…

### LINKED ISSUE #19: 如果 npm package 的 package.json 里有 exports， 并且里面没有 ./package.json key，build.js 就会报错
如果 npm package 的 package.json 里有 exports， 并且里面没有 ./package.json key，build.js 就会报错
```js
import.meta.resolve(`${module}/package.json`) // 如果 package.json 里有exports，并且没有包含 `./package.json` 就会报错
```

```console
node:internal/modules/esm/resolve:304
  return new ERR_PACKAGE_PATH_NOT_EXPORTED(
         ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './package.json' is not defined by "exports" in D:\Code\Tmp\xxx\node_modules\lodash\package.json imported from D:\Code\Tmp\xxx\build.js
    at exportsNotFound (node:internal/modules/esm/resolve:304:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:651:9)
    at packageResolve (node:internal/modules/esm/resolve:837:14)
    at moduleResolve (node:internal/modules/esm/resolve:927:18)
    at defaultResolve (node:internal/modules/esm/resolve:1157:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:390:12)
    at ModuleLoader.resolveSync (node:internal/modules/esm/loader:372:17)
    at Object.resolve (node:internal/modules/esm/initialize_import_meta:30:25)
    at buildComponentLibrary (file:///D:/Code/Tmp/xxx/build.js:101:25)
    at scanDependencies (file:///D:/Code/Tmp/xxx/build.js:158:21) {
```

具体报错是这里：
https://github.com/nodejs/node/blob/9db6327af3272c4125d20c078010600629506a26/lib/internal/modules/esm/resolve.js#L586

### PR DESCRIPTION
**Pull request by CGQAQ**
….js 报错问题

refs:
- https://antfu.me/posts/get-package-root
- https://github.com/antfu-collective/local-pkg

TLDR: 当 resolve `module/package.json` 抛异常时，会走 [fallback](https://github.com/antfu-collective/local-pkg/blob/c10d34f1dbce5fdcf2723c6b1c6636e3797b9fdd/src/index.ts#L99-L114)

See: 
- [1](https://github.com/antfu-collective/local-pkg/blob/c10d34f1dbce5fdcf2723c6b1c6636e3797b9fdd/src/index.ts#L59-L65)
- [2](https://github.com/antfu-collective/local-pkg/blob/c10d34f1dbce5fdcf2723c6b1c6636e3797b9fdd/src/index.ts#L116-L131)

fixes: #19

### CODE REVIEW

**Comment by CGQAQ** [comment]
@yangmingshan 请看一下
